### DISCUSSION TITLE
1.11 桶排序

### DISCUSSION THREAD

**Discussion by giscus** [itcharge/AlgoNote#111] [Comments]
# 1.11 桶排序

1. 桶排序算法思想 桶排序（Bucket Sort）基本思想： 将待排序数组中的元素分散到若干个「桶」中，然后对每个桶中的元素再进行单独排序。 2. 桶排序算法步骤 1. 确定桶的数量：根据待排序数组的值域范围，将数组划分为 $k$ 个桶，每个桶可以看做是一个范围区间。 2. 分配元素：遍历待排序数组元素，将每个元素根据大小分配到对应的桶中。 3. ...

[https://algo.itcharge.cn/01_array/01_11_array_bucket_sort/](https://algo.itcharge.cn/01_array/01_11_array_bucket_sort/)

**Comment by OnlyBrownAnt** [comment]
博主我看你这儿的实现是通过定义桶容量bucket_size来处理得到桶数量bucket_count。那如果定义桶的区间范围来得到桶数量可以吗？比如桶数量 = 向上取整(最大值 - 最小值 / 区间范围)，因为一般来说数据不一定是均匀的，很可能会存在某个区间数据元素较多的情况，可能这个桶还得进行容量扩容。

**Comment by Wan-Yifei** [comment]
桶排序和前文的快速排序等算法很多都支持平行处理，不如平行对所有桶同时进行排序。作者可以对支持平行计算的算法都提一句，这也算是个算法优势。

**Reply by Wan-Yifei** [reply]
勘误： “不如平行” -> “比如平行”
